portal/src/main/webapp/WEB-INF/jsp/step4_json.jsp

```diff
@@ -59,27 +59,36 @@
         <button id="toggle_mutsig_dialog" onclick="promptMutsigTable(); return false;" style="font-size: 1em;">Select From Recurrently Mutated Genes (MutSig)</button>
         <button id="toggle_gistic_dialog_button" onclick="Gistic.UI.open_dialog(); return false;" style="font-size: 1em; display: none;">Select Genes from Recurrent CNAs (Gistic)</button>
     </div>
+    
 
     <script type="text/javascript">
         $(document).ready(function() {
             GeneSymbolValidator.initialize();
+            OqlMenu.initialize();
         });
     </script>
+    
+    <div contenteditable="true" id="gene_list" placeholder="Enter HUGO Gene Symbols or Gene Aliases" title="Enter HUGO Gene Symbols or Gene Aliases" spellcheck="false"><%
+        if (localGeneList != null && localGeneList.length() > 0) {
+	        String geneListWithSemis =
+			        org.mskcc.cbio.portal.oncoPrintSpecLanguage.Utilities.appendSemis(localGeneList);
```

Comment:
What are Semis? We're aiming to phase out all backend OQL code so it would be better to do this on the frontend.